sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiver.cs

```diff
@@ -242,6 +242,21 @@ protected PartitionReceiver()
             OwnsConnection = false;
         }
 
+        /// <summary>
+        ///   Retrieves information about the partition this client is associated to, including elements that describe the
+        ///   available events in the partition event stream.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">An optional <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <returns>The set of information for the associated partition under the Event Hub this client is associated with.</returns>
+        ///
+        public virtual Task<PartitionProperties> GetPartitionPropertiesAsync(CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotClosed(IsClosed, nameof(PartitionReceiver));
+            return Connection.GetPartitionPropertiesAsync(PartitionId, RetryPolicy, cancellationToken);
+        }
```

Comment:
Once again, I'm copying the behavior from other clients. @jsquire I remember you mentioning it was preferred to make the method `async` and calling `await` on the connection call (at least in that specific `EventProcessor` method). Is this something we want to do here as well? I can't really remember the reasoning behind it.

Reply:
We would need to update other clients if that's the case.